src/main/java/io/stargate/sgv2/jsonapi/service/cqldriver/CQLSessionCache.java

```diff
@@ -151,18 +151,54 @@ private CqlSession getNewSession(SessionCacheKey cacheKey) {
       }
       return builder.build();
     } else if (ASTRA.equals(databaseConfig.type())) {
-      return new TenantAwareCqlSessionBuilder(cacheKey.tenantId())
-          .withAuthCredentials(
-              TOKEN, Objects.requireNonNull(((TokenCredentials) cacheKey.credentials()).token()))
-          .withLocalDatacenter(operationsConfig.databaseConfig().localDatacenter())
-          .withClassLoader(Thread.currentThread().getContextClassLoader())
-          .withApplicationName(APPLICATION_NAME)
-          .withConfigLoader(loader)
-          .build();
+      CqlSession cqlSession =
+          new TenantAwareCqlSessionBuilder(cacheKey.tenantId())
+              .withAuthCredentials(
+                  TOKEN,
+                  Objects.requireNonNull(((TokenCredentials) cacheKey.credentials()).token()))
+              .withLocalDatacenter(operationsConfig.databaseConfig().localDatacenter())
+              .withClassLoader(Thread.currentThread().getContextClassLoader())
+              .withApplicationName(APPLICATION_NAME)
+              .withConfigLoader(loader)
+              .build();
+      if (!isAstraSessionValid(cqlSession, cacheKey.tenantId())) {
+        throw new UnauthorizedException(
+            "Unauthorized to access tenant %s's data".formatted(cacheKey.tenantId()));
```

Comment:
don't use the word "tenant" in anything user facing.  Also don't put the ID in the message.  There error message here should be no different from the error thrown when a bad token is used.